File: domain/src/main/java/no/nav/melosys/domain/mottatteopplysninger/data/OppholdUtland.java
package no.nav.melosys.domain.mottatteopplysninger.data;

import java.util.ArrayList;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnore;
import no.nav.melosys.domain.ErPeriode;
import no.nav.melosys.domain.HarPeriode;

/**
 * Opplysninger om opphold i utland
 */
public class OppholdUtland implements HarPeriode {
    public List<String> oppholdslandkoder = new ArrayList<>();
    public Periode oppholdsPeriode = new Periode();
    public String studentFinansieringKode;
    public String studentSemester;
    public Boolean ektefelleEllerBarnINorge;

    @Override
    @JsonIgnore
    public ErPeriode getPeriode() {
        return oppholdsPeriode;
    }
}

File: domain/src/main/kotlin/no/nav/melosys/domain/mottatteopplysninger/data/OppholdUtland.kt
package no.nav.melosys.domain.mottatteopplysninger.data

import com.fasterxml.jackson.annotation.JsonIgnore
import no.nav.melosys.domain.ErPeriode
import no.nav.melosys.domain.HarPeriode


/**
 * Opplysninger om opphold i utland
 */
class OppholdUtland : HarPeriode {
    var oppholdslandkoder: List<String> = ArrayList()
    var oppholdsPeriode: Periode? = null
    var studentFinansieringKode: String? = null
    var studentSemester: String? = null
    var ektefelleEllerBarnINorge: Boolean? = null

    @JsonIgnore
    override fun getPeriode(): ErPeriode? = oppholdsPeriode?.tilErPeriode()
}
